src/node/stereo_panner.rs

```diff
@@ -148,9 +207,10 @@ impl StereoPannerNode {
 
 /// `StereoPannerRenderer` represents the rendering part of `StereoPannerNode`
 struct StereoPannerRenderer {
-    /// The position of the input in the output’s stereo image.
+    /// Position of the input in the output’s stereo image.
     /// -1 represents full left, +1 represents full right.
     pan: AudioParamId,
+    buffer: [[f32; 2]; 128],
```

Comment:
You are using an temp buffer because rust won't let you get two `channel_data_mut` at the same time? Let's build a helper to fix that

Reply:
No actually, that was kind of how it was made in the first place, i.e.

```
out_data[0][sample_idx] = left;
out_data[1][sample_idx] = right;
```

But in terms of perf this is less good because (at least as I understand it) your writing head has to switch constantly between two memory regions, while with this intermediary buffer, everything is well aligned (that's a bit counterintuitive but that was the idea in the `AudioBufferSourceNode` and `DelayNode ` that were using the now disappeared `RenderQuantum::set_channels_values_at` (https://github.com/orottier/web-audio-api-rs/commit/9a3993054fdf621b3acc494df3c7170a82c3a718#diff-717673885f123988d3c7e106e512029ce25a5aed857fac7e31e4cc04352ce946)

Happy to have another try if you think it worth it but I'm pretty sure the perfs will go down, I really tested a lot of strategies

Reply:
Actually, the more I think about it the more I think it's related to this discussion between planar and interleaved layouts. I slowly start to understand why interleaved is better / more simple then... (given that you deal with fixed number of channels)

Reply:
Definitely something to reconsider yes. I really appreciate the amount of work and thoughts you are putting in this stuff!